odoo/cli/module.py

```diff
@@ -0,0 +1,163 @@
+            self.uninstall_parser,
+            self.force_demo_parser,
+            self.upgrade_parser,
+        ):
+            parser.add_argument(
+                '-c', '--config', dest='config',
+                help="use a specific configuration file")
+            parser.add_argument(
+                '-d', '--database', dest='db_name', default=None,
+                help="database name, connection details will be taken from the config file")
+
+        self.install_parser.add_argument(
+            'modules', nargs='+', metavar='MODULE',
+            help="modules to be installed")
+        self.uninstall_parser.add_argument(
+            'modules', nargs='+', metavar='MODULE',
+            help="modules to be uninstalled")
+        self.upgrade_parser.add_argument(
+            'modules', nargs='+', metavar='MODULE',
+            help="modules to be upgraded")
+        self.upgrade_parser.add_argument(
+            'modules', nargs='+', metavar='MODULE',
+            help="modules to be upgraded")
+        self.force_demo_parser.add_argument(
+            'modules', nargs='*', metavar='MODULE',
+            help="modules to have demo data loaded")
+
+    def run(self, cmdargs):
+        parsed_args = self.parser.parse_args(args=cmdargs)
+
+        config_args = []
+        if parsed_args.config:
+            config_args += ['-c', parsed_args.config]
+        if parsed_args.db_name:
+            config_args += ['-d', parsed_args.db_name]
+        config.parse_config(config_args, setup_logging=True)
+
+        db_names = config['db_name']
+        if not db_names or len(db_names) > 1:
+            self.parser.error("Please provide a single database in the config file")
+        parsed_args.db_name = db_names[0]
+
+        match parsed_args.subcommand:
+            case 'install':
+                self._install(parsed_args)
+            case 'uninstall':
+                self._uninstall(parsed_args)
+            case 'upgrade':
+                self._upgrade(parsed_args)
+            case 'force-demo':
+                self._force_demo(parsed_args)
+
+    def _check_modules_on_disk(self, parser, parsed_args):
+        """ Check modules on disk before starting the Db environment """
+        initialize_sys_path()
+        modules = set(parsed_args.modules)
+        valid_modules = {
+            module
+            for module in modules
+            if get_module_path(module, display_warning=False)
+        }
+        if invalid_modules := modules - valid_modules:
+            _logger.warning("Some of the modules weren't found on the filesystem: %s", ", ".join(invalid_modules))
+        if not valid_modules:
+            parser.error(f"No modules to {parsed_args.subcommand}")
+        return valid_modules
+
+    def _get_modules(self, env, domain, action):
+        """ Get modules from the db, depending on the state """
+        Module = env['ir.module.module']
+        Module.update_list()
+        return Module.search(domain)
```

Comment:
What's the purpose of `action`? Do you want to log something?

Reply:
Old, dead code, removing.